Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/AzureMvpModel.java

```diff
@@ -140,9 +142,15 @@ public ResourceGroup getResourceGroupBySubscriptionIdAndName(String sid, String
      * @param sid subscription Id
      * @return List of Location instances
      */
-    public List<Location> listLocationsBySubscriptionId(String sid) {
+    public List<Location> listLocationsBySubscriptionId(String sid) throws Exception {
+        List<Location> locations = new ArrayList<>();
         Subscription subscription = getSubscriptionById(sid);
-        return subscription.listLocations();
+        try {
+            locations.addAll(subscription.listLocations());
+        } catch (Exception e) {
+            throw new Exception(CANNOT_GET_LOCATIONS);
```

Comment:
we should wrap the origin exception and add our message. Instead of discard th original exception